cmd/hummingbird/main.go

```diff
@@ -204,144 +198,168 @@ func ProcessControlCommand(serverCommand func(name string, args ...string)) {
 
 	switch flag.Arg(1) {
 	case "proxy", "object", "object-replicator", "object-auditor", "container", "container-replicator", "account", "account-replicator":
-		serverCommand(flag.Arg(1), flag.Args()[2:]...)
+		if err := serverCommand(flag.Arg(1), flag.Args()[2:]...); err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+	case "main":
+		exc := 0
+		for _, server := range []string{"proxy", "object", "container", "account"} {
+			if err := serverCommand(server); err != nil {
+				fmt.Fprintln(os.Stderr, server, ":", err)
+				exc = 1
+			}
+		}
+		os.Exit(exc)
 	case "all":
+		exc := 0
 		for _, server := range []string{"proxy", "object", "object-replicator", "object-auditor",
 			"container", "container-replicator", "account", "account-replicator"} {
-			serverCommand(server)
+			if err := serverCommand(server); err != nil {
+				fmt.Fprintln(os.Stderr, server, ":", err)
+				exc = 1
+			}
 		}
+		os.Exit(exc)
 	default:
 		flag.Usage()
 	}
 }
 
-func main() {
-	common.SetRlimits()
-	rand.Seed(time.Now().Unix())
-
-	/* sub-command flag parsers */
+func init() {
+	rand.Seed(time.Now().UTC().UnixNano())
```

Comment:
I think moving this to init() has caused test failures than otherwise relied on a static rand seed. Probably should fix those tests rather than moving this. common/ring/ring_test.go TestGetNodes is the one that blew up.